lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
The size of the buffer will always be equal to the size of the data inside it because it is coming from 'data' event on the socket, so there is never enough space to copy the second buffer inside it. And you can't just resize the buffer because it is just a view over the ArrayBuffer underneath, you can only create a new one.
I still can't understand how you want this to be implemented.